### ISSUE TITLE
BoxInBox: When recovering from a checkpoint, variables are invalid in boundaries and ghost zones.

### ISSUE THREAD

**Issue by acwen11** [eschnett/CarpetX#168]
This is the resulting error:
```
  -> CallFunction iteration 420 CCTK_POSTSTEP: BoxInBox::EstimateError checking input: Array "BOXINBOX::active[0]" is invalid on time level 0; required
  The interior is valid.
  The outer boundary is valid.
  The ghost zones are valid.
, found
  The interior is valid because: read from openPMD file.
  The outer boundary is invalid because: read from openPMD file.
  The ghost zones are invalid because: read from openPMD file.
```

**Comment by acwen11** [comment]
[qc0.txt](https://github.com/eschnett/CarpetX/files/12420437/qc0.txt)
[qc0_PTonly.txt](https://github.com/eschnett/CarpetX/files/12420438/qc0_PTonly.txt)

(It seems like I am not able to attach any files with a `.par` extension)
`qc0.par` adds `BoxInBox` refinement, horizon finding and quasilocal measures to the qc0 par file in `Z4c` and fails to save checkpoints due to integer and complex types not being supported. `qc0_PTonly.par` only adds `BoxInBox`, saves checkpoints fine, but fails to load them after grid arrays are flagged as invalid on boundaries and ghost zones.

**Comment by eschnett** [comment]
Closed via #209.
